src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -576,7 +579,7 @@ private void OnEnvDTEProjectRemoved(EnvDTE.Project envDTEProject)
 
         private void OnEnvDTEProjectAdded(EnvDTE.Project envDTEProject)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
```

Comment:
can try EndToEnd test, we have lots of "CreateSolution then AddProject" there, can't repro every time, depends on timing